src/main/scala/io/getclump/ClumpFetcher.scala

```diff
@@ -1,10 +1,10 @@
 package io.getclump
 
-import scala.collection.mutable.{Map => MutableMap}
+import scala.collection.mutable
 
 private[getclump] final class ClumpFetcher[T, U](source: ClumpSource[T, U]) {
 
-  private[this] val fetches = MutableMap[T, Promise[Option[U]]]()
+  private[this] val fetches = mutable.LinkedHashMap[T, Promise[Option[U]]]()
```

Comment:
The map is used only as a temporary cache and we don't iterate over it. I can't see why it is necessary.